x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/text_expansion_errors.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import { EuiCallOut, EuiLink } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { HttpError } from '../../../../../../../common/types/api';
+import { getErrorsFromHttpResponse } from '../../../../../shared/flash_messages/handle_api_errors';
+import { HttpLogic } from '../../../../../shared/http';
+
+import { ML_NOTIFICATIONS_PATH } from '../../../../routes';
+
+export const TextExpansionErrors = ({
```

Comment:
Suggestion: this component is pretty complex with knowing which title to display. Have you considered moving this logic up into the logic file, passing in a single error object?

It would lead to cleaner display component and easier testing. The same error object can be then shared across components.

Reply:
Refactored, see [commit](https://github.com/elastic/kibana/pull/156273/commits/bb605dbbb5fb9d50669bcffd63ceb1f1a7dfc517). Note I'm not introducing a new type for the error, just a compound object.